lib/reservation.rb

```diff
@@ -0,0 +1,33 @@
+require "date"
+
+module Hotel
+  class Reservation
+    attr_reader :room, :start_date, :end_date
+
+    def initialize(room:, start_date:, end_date:)
+      @room = room
+
+      @start_date, @end_date = parse_dates(start_date, end_date)
+    end
+
+    def calculate_cost
+      total_cost = ((@end_date - @start_date) - 1) * @room.cost_per_night
+      return total_cost
+    end
```

Comment:
I like that you use a method for this rather than storing it in an instance variable. That way if the number of nights or cost per night changes, the cost will always be correct.